multiqc/modules/verifybamid/verifybamid.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from VerifyBAMID """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+from multiqc import config
+from multiqc.plots import table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+     module class, parses stderr logs.
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='verifyBAMID', anchor='verifybamid',
+        href='https://genome.sph.umich.edu/wiki/VerifyBamID',
+        info="detect sample contamination and/or sample swap")
+        
+        # dictionary to hold all data for each sample
+        self.verifybamid_data = dict()
+
+        # for each file ending in self.SM
+        for f in self.find_log_files('verifybamid/selfsm'):
+            # add data source to multiqc_sources.txt 
+            self.add_data_source(f)
+            # pass the file to function self.parse_selfsm to parse file
+            parsed_data = self.parse_selfsm(f)
+            # if a result was returned
+            if parsed_data is not None:
+                # for each sample extracted from the file
+                for s_name in parsed_data:
+                    # add the sample as a key to the verifybamid_data dictionary and the dictionary of values as the value
+                    self.verifybamid_data[s_name] = parsed_data[s_name]
+        
+        # Filter to strip out ignored sample names
+        self.verifybamid_data = self.ignore_samples(self.verifybamid_data)
+
+        if len(self.verifybamid_data) == 0:
+            raise UserWarning
+
+        # print number of verifyBAMID reports found and parsed
+        log.info("Found {} reports".format(len(self.verifybamid_data)))
+
+        # Write parsed report data to a file	
+        self.write_data_file(self.verifybamid_data, 'multiqc_verifybamid')
+
+        # add to General Stats Table
+        self.verifybamid_general_stats_table()
+
+        # add section with the values from the verify BAM ID output
+        self.verifybamid_table()
+
+
+    def parse_selfsm(self, f):
+        """ Go through selfSM file and create a dictionary with the sample name as a key, """
+        #create a dictionary to populate from this sample's file
+        parsed_data = dict()
+        # set a empty variable which denotes if the headers have been read
+        headers = None
+        # for each line in the file
+        for l in f['f'].splitlines():
+                # split the line on tab
```

Comment:
It doesn't matter, but my OCD can't help but notice that you have double indentation in this block 🤓